rs/registry/canister/src/mutations/node_management/common.rs

```diff
@@ -226,6 +226,13 @@ fn get_key_family<T: prost::Message + Default>(
     registry: &Registry,
     prefix: &str,
 ) -> Vec<(String, T)> {
+    get_key_family_iter(registry, prefix).collect()
+}
+
+pub(crate) fn get_key_family_iter<'a, T: prost::Message + Default>(
```

Comment:
NCR: why not put this in

```
impl<'a> Registry {
    pub fn key_family<'a, T: prost::Message + Default>(&'a self, &'a str) -> impl Iterator<Item = (String, T)> + 'a { 
        ...
    }
}
```

Also, I don't know if we need to write `get_` or `_iter`, the rust standard library doesn't do this and I think it's clear enough what it's doing without it (e.g. to get an iterator over the values of a hashmap you just do `mymap.values()`).

Reply:
There's already a "get_key_family" which just returns all the key/value pairs, and I wanted to differentiate between that and a function that returns an iterator.

The main change here is to stop iterating over all the registry keys like we currently do, which is quite expensive.